platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/fingerprint/ConfigurationCacheFingerprintWriter.kt

```diff
@@ -923,6 +932,97 @@ class ConfigurationCacheFingerprintWriter(
+    ) {
+        if (isInputTrackingDisabled()) {
+            return
+        }
+        if (gradlePropertiesByPrefix.computeIfAbsent(propertyScope) { newConcurrentHashSet() }.add(prefix)) {
+            sink().write(
+                ConfigurationCacheFingerprint.GradlePropertiesPrefixedBy(
+                    propertyScope,
+                    prefix,
+                    snapshot
+                )
+            )
+            reportGradlePropertiesByPrefixInput(propertyScope, prefix)
+        }
+    }
+
+    override fun onGradlePropertyAccess(
+        propertyScope: GradlePropertiesListener.PropertyScope,
+        propertyName: String,
+        propertyValue: Any?
+    ) {
+        if (isInputTrackingDisabled()) {
+            return
+        }
+        if (gradleProperties.computeIfAbsent(propertyScope) { newConcurrentHashSet() }.add(propertyName)) {
+            sink().write(
+                ConfigurationCacheFingerprint.GradleProperty(
+                    propertyScope,
+                    propertyName,
+                    propertyValue
+                )
+            )
+            reportGradlePropertyInput(propertyScope, propertyName)
+        }
+    }
+
+    private
+    fun reportGradlePropertyInput(
+        propertyScope: GradlePropertiesListener.PropertyScope,
+        propertyName: String,
+        consumer: String? = null
+    ) {
+        if (propertyName.startsWith("org.gradle.")) {
+            // Don't include builtin Gradle properties in the report
+            return
+        }
+        val location = locationFor(consumer)
+        if (location === PropertyTrace.Unknown || location === PropertyTrace.Gradle) {
+            // Don't include property accesses coming from the Gradle runtime itself (e.g., version, group, buildDir, etc.)
+            return
+        }
+        reportInput(scopedLocation(propertyScope, location), null) {
+            text("gradle property ")
+            reference(propertyName)
+        }
+    }
+
+    private
+    fun reportGradlePropertiesByPrefixInput(
+        propertyScope: GradlePropertiesListener.PropertyScope,
+        prefix: String,
+        consumer: String? = null
+    ) {
+        val location = locationFor(consumer)
+        if (location === PropertyTrace.Unknown || location === PropertyTrace.Gradle) {
+            // Don't include property accesses coming from the Gradle runtime itself (e.g., sysProp.*)
+            return
+        }
+        reportInput(scopedLocation(propertyScope, location), null) {
+            text("gradle property ") // To avoid introducing a separate subtree in the report
+            reference("$prefix.*")
+        }
+    }
+
+    private
+    fun scopedLocation(
+        propertyScope: GradlePropertiesListener.PropertyScope,
+        location: PropertyTrace
+    ) = PropertyTrace.Project(
+        path = when (propertyScope) {
+            is GradlePropertiesListener.PropertyScope.Project -> propertyScope.projectIdentity.buildTreePath.toString()
```

Comment:
💅  Better to use an explicit method than a `toString()` 

```suggestion
            is GradlePropertiesListener.PropertyScope.Project -> propertyScope.projectIdentity.buildTreePath.path
```

Reply:
Right, done.